### PR TITLE
add an example for instancedmesh selection

### PR DESCRIPTION
**Pull request by chiguaboy**
Related issue: #22399

**Description**

- add an example for instancedmesh selection with transform #22399
- add **dispose** for selectionhelper to clean up 
- functional selection to switch between selection/OrbitControl

### CODE REVIEW

**Comment by Mugen87** [comment]
TBH, I don't think we need a new example for `SelectionBox`. A single one should be sufficient even if don't test with all supported object types.

I suggest you make a new PR and only add the `dispose()` routine for `SelectionHelper`.

**Comment by chiguaboy** [comment]
> TBH, I don't think we need a new example for `SelectionBox`. A single one should be sufficient even if don't test with all supported object types.
> 
> I suggest you make a new PR and only add the `dispose()` routine for `SelectionHelper`.

> My bad expression.I did not mean to add one example for each object type.
> The example `misc_boxselection` seems to be minimal, but i think one example considering more relative basic usage like mouse control  before/after selection will be more sufficient for developers ,or just too wordy😆

**Comment by Mugen87** [comment]
In general we try to keep the examples compact and not overload them with features. That's the reason why I have suggested to just focus on `SelectionHelper`. Introducing `dispose()` for this class is indeed a valuable addition!

**Comment by chiguaboy** [comment]
> In general we tend to keep the examples compact and not overload them with features. That's the reason why I have suggested to just focus on `SelectionHelper`. Introducing `dispose()` for this class is indeed a valuable addition!

> got it
